### PR TITLE
MM-21300 : Migrate /benchmark e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
Migrate /benchmark tests to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21300

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by nevyangelova — CHANGES_REQUESTED** [review]
@sinansonmez i don't see the changes to the message_spec file

**Comment by sinansonmez** [comment]
Yes, changing the file extension from `js` to `ts` was enough. There was no need to add any type.

**Comment by nevyangelova** [comment]
@sinansonmez sorry for being nit-picky, there are some function parameters and return types that can be typed, do you mind taking a look? 

After taking a look at most of our specs we don't yet have a ts convention for our cypress tests. So I am going to be more lenient with the review as well :)

**Review by nevyangelova — APPROVED** [review]

File: e2e/cypress/tests/support/win.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// ***************************************************************
+
+declare namespace Cypress {
+    interface ApplicationWindow {
+
+        /**
+     * Reset all tracked selectors
+     * @example
+     *   win.resetTrackedSelectors();
+     */
```
Comment (saturninoabril) [inline e2e/cypress/tests/support/win.d.ts:23]: Please fix indentation. Unfortunately, this seemed to be not covered by our lint setup.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @sinansonmez! LGTM, except for one request.

File: e2e/cypress/tests/support/win.d.ts (hunk 1 above)
Comment (sinansonmez) [reply e2e/cypress/tests/support/win.d.ts]: Done, thanks

**Review by saturninoabril — APPROVED** [review]
Thanks @sinansonmez!

**Pull request merged**